src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -67,27 +67,34 @@ static_assert(LWIP_TCPIP_CORE_LOCKING, "CHIP requires config LWIP_TCPIP_CORE_LOC
 namespace chip {
 namespace Inet {
 
-namespace {
-/**
- * @brief
- * RAII locking for LwIP core to simplify management of
- * LOCK_TCPIP_CORE()/UNLOCK_TCPIP_CORE() calls.
- */
-class ScopedLwIPLock
+EndpointQueueFilter * UDPEndPointImplLwIP::sQueueFilter = nullptr;
+
+static void LockChipStack()
+{
+    // DeviceLayer::PlatformMgr().LockChipStack();
+}
+
+static void UnlockChipStack()
 {
-public:
-    ScopedLwIPLock() { LOCK_TCPIP_CORE(); }
-    ~ScopedLwIPLock() { UNLOCK_TCPIP_CORE(); }
+    // DeviceLayer::PlatformMgr().UnlockChipStack();
+}
+
+struct TCPIPCallContext
+{
+    struct tcpip_api_call_data call;
+    std::function<err_t()> fn;
 };
-} // anonymous namespace
 
-EndpointQueueFilter * UDPEndPointImplLwIP::sQueueFilter = nullptr;
+err_t RunOnTCPIP(std::function<err_t()> fn)
+{
+    TCPIPCallContext ctx = { .fn = fn };
+    return tcpip_api_call([](struct tcpip_api_call_data * call) { return reinterpret_cast<TCPIPCallContext *>(call)->fn(); },
```

Comment:
May we use the public API `tcpip_callback` instead of `tcpip_api_call` here?

Reply:
not quite: `tcpip_callback` just schedules a function to run on the TCPIP task, `tcpip_api_call` also waits for it to run and returns the result.
it's a fair point that we should not be using internal APIs, so i implemented `RunOnTCPIP` in the spirit of `tcpip_api_call` but without actually using it.